lib/distribute.c

```diff
@@ -546,13 +612,15 @@ int config_write_distribute(struct vty *vty)
 /* Clear all distribute list. */
 void distribute_list_reset()
 {
-	hash_clean(disthash, (void (*)(void *))distribute_free);
+	hash_clean(dist_ctx.disthash, (void (*)(void *))distribute_free);
 }
 
 /* Initialize distribute list related hash. */
-void distribute_list_init(int node)
+void *distribute_list_init(int node,
```

Comment:
I think you need to split this function into two:
* `distribute_list_init(int node)`: install the distribute-list commands under the given vty node.
* `distribute_ctx_create(struct vrf *vrf)`: create a distribute-list container and return a pointer to it.

Additionally, `distribute_list_reset()` can be renamed to `distribute_ctx_delete()` and accept a `struct distribute_ctx` pointer as a parameter.